meshroom/ui/qml/ImageGallery/ImageGallery.qml

```diff
@@ -201,6 +200,19 @@ Panel {
                 }
             }
 
+            // Update grid item when corresponding thumbnail is computed
+            Connections {
+                target: ThumbnailCache
+                function onThumbnailCreated(imgSource) {
```

Comment:
send the index in `ThumbnailCache.thumbnail(root.source);` and emit it in the signal to retrieve it here and avoid the quadratic loop here.
```
function onThumbnailCreated(imgSource, index) {
    if(index => grid.count)
        return
    let item = grid.itemAtIndex(i);
    if(!item)
        return
    return item.updateThumbnail();
}
```